infra/notifications/src/app.js

```diff
@@ -165,68 +165,107 @@ app.post('/', async (req, res) => {
-      deviceToken: mobileRegister.deviceToken
+  if (!mobileRegister.ethAddress) {
+    return res.status(400).send({ errors: ['Invalid Eth address'] })
+  }
+  // By convention all Eth addresses are stored lower cased in the DB.
+  mobileRegister.ethAddress = mobileRegister.ethAddress.toLowerCase()
+
+  if (mobileRegister.deviceToken) {
+    // See if a row already exists for this (Eth address, device token) pair.
+    const registryRow = await MobileRegistry.findOne({
+      where: {
+        ethAddress: mobileRegister.ethAddress,
+        deviceToken: mobileRegister.deviceToken
+      }
+    })
+
+    if (!registryRow) {
+      // Nothing exists, create a new row.
+      await MobileRegistry.create(mobileRegister)
+      res.sendStatus(201)
+      logger.debug('Added new mobile device to registry: ', req.body)
+    } else {
+      // Update existing row since permissions might have changed.
+      await MobileRegistry.upsert(mobileRegister)
+      res.sendStatus(200)
+      logger.debug('Updated mobile device registry: ', req.body)
     }
-  })
-
-  if (!registryRow) {
-    // Nothing exists, create a new row
-    logger.debug('Adding new mobile device to registry: ', req.body)
-    registryRow = await MobileRegistry.create(mobileRegister)
-
-    // Record the mobile account creation in the growth_event table.
-    await GrowthEvent.insert(
-      logger,
-      1,
-      mobileRegister.ethAddress,
-      GrowthEventTypes.MobileAccountCreated,
-      mobileRegister.deviceToken,
-      { deviceType: mobileRegister.deviceType },
-      new Date()
-    )
-
-    res.sendStatus(201)
-  } else {
-    // Row exists, permissions might have changed, update if required
-    logger.debug('Updating mobile device registry: ', req.body)
-    registryRow = await MobileRegistry.upsert(mobileRegister)
-    res.sendStatus(200)
   }
+
+  // Record the mobile account creation in the growth_event table.
+  // Notes:
+  //  - deviceToken may be null. Not an issue since it is just informational
+  //    metadata that is not used.
+  //  - The insert method is idempotent. It checks for existing rows before
+  //    inserting, so it's alright to call it every time /mobile/register
+  //    gets executed.
+  await GrowthEvent.insert(
+    logger,
+    1,
+    mobileRegister.ethAddress,
+    GrowthEventTypes.MobileAccountCreated,
+    mobileRegister.deviceToken,
+    { deviceType: mobileRegister.deviceType },
+    new Date()
+  )
+  logger.debug(
+    `Recorded mobile account creation for ${mobileRegister.ethAddress} in growth system.`
+  )
 })
 
+/**
+ * Unregisters a device for push notifications.
+ */
 app.delete('/mobile/register', async (req, res) => {
-  logger.info('Call to delete mobile registry endpoint')
+  const mobileRegister = {
+    ethAddress: _.get(req.body, 'eth_address', null),
+    deviceToken: _.get(req.body, 'device_token', null)
+  }
+  logger.info(`DELETE /mobile/register for ${mobileRegister.ethAddress}`)
+
+  if (!mobileRegister.ethAddress) {
```

Comment:
might be worth doing a little more validation here, but not a deal breaker.

Reply:
Good point. Added address validation.